### ISSUE TITLE
scikit-image sprint at BIDS May 28 - Jun 1

### ISSUE THREAD

**Issue by jni** [scikit-image/scikit-image#3086]
## Description
This issue collects ideas and issues for the scikit-image side of the joint scikit-learn/scikit-image/dask sprint at UC Berkeley. 

- work on the parallelization of some algorithms to improve performance, focussing maybe first on the case of machines with several cores (rather than a distributed architecture, which most users don't use I guess). This would need some benchmarking and we could test several solutions such as dask (with Matt Rocklin, we could test and maybe improve our apply_parallel) as well as joblib and its different backends (with Gaël and Olivier). Emma can provide some tomography datasets for this, it would be better if we had a machine with 10-20 cores to ssh on for benchmarking, maybe with AWS? Or a BIDS machine?
- discussing all [GitHub "needs decision" PRs/issues](https://github.com/scikit-image/scikit-image/labels/status%3A%20needs%20decision). 
- get nD transforms, based on NumPy array coordinates, working. (with @kne42)
- figure out whether we can reliably distribute Numba, and if so, do we want to get that dependency.
- implement flood-fill with support for lowlevelcallables (see https://github.com/scikit-image/scikit-image/issues/2876#issuecomment-385241507)
- fix data types and ranges. (See [#3009](https://github.com/scikit-image/scikit-image/issues/3009).)
- discuss and implement possible matmul interface for transforms instead of add. (with @hmaarrfk)
- talk about release schedules, governance, funding, leadership, outreach, and a few more general topics. (Though @jni thinks these things should happen in the evenings over dinner etc. ;)
- deprecate the skimage viewer to move it to its own repo.

Stuff that we want to happen *before* the sprint:
- [x] announce the sprint on the mailing list to see whether anyone wants to join remotely
- [x] release 0.14
- [ ] get airspeed velocity working (though maybe this needs to happen *before* the sprint anyway)
- [ ] draw up scikit-image 1.0 roadmap (including emailing mailing list for community discussion)

Note also [this repo](https://github.com/scisprints/2018_05_sklearn_skimage_dask) for ideas for *joint* work with sklearn and dask teams. I suspect @jakirkham will be the ringleader for that. =)

**Comment by stefanv** [comment]
- Decide on new core members to invite
- Discuss user survey & usage exhibition on website
- Discuss funding opportunities
- Add linking to objects & functions in gallery (maybe this can just go into an issue); should we provide mybinder links for documentation?
- Cross-language usage examples in documentation (skimage + ImageJ in microscopy, e.g.)? 
- Figure out how to include long form errors in code with appropriate formatting 
- Consider nightly builds for Windows users?
- Paper v2: interest, necessity, process

**Comment by hmaarrfk** [comment]
I would like to work on Image stitching and registration.

**Comment by stefanv** [comment]
Excellent!  Also see http://imreg-dft.readthedocs.io/en/latest/,
https://github.com/pyimreg/imreg,
https://github.com/stefanv/ned_imagepairs,
https://github.com/SuperElastix/SuperElastix

**Comment by hmaarrfk** [comment]
@stefanv, I'll have a look at those links. I should have specified that I am particularly interested in the problem for microscopy purposes, where I believe that the 2D assumption is fair.

**Comment by stefanv** [comment]
In that case: http://nipy.org/dipy/examples_index.html#registration

**Comment by stefanv** [comment]
Consider removing `skimage.future`, or how to make sure packages graduate out.

**Comment by jni** [comment]
Yeah we need a process for it. Things should not spend more than two releases there. So I think the idea should be: if we're putting it in `future`, add the graduation to the to-do.

**Comment by stefanv** [comment]
Dask: can we attach hints to various functions on how they would be computed in parallel (e.g., if a kernel is involved, what is the kernel width), and then have one utility function that does it? /cc @mrocklin

**Comment by soupault** [comment]
Could someone summarize the outcomes and the results of the work made on the provided topics?

**Comment by sciunto** [comment]
@soupault Done in #3263 I'm closing this issue.
